consensus/state.go

```diff
@@ -1899,7 +1899,7 @@ func (cs *State) defaultSetProposal(proposal *types.Proposal) error {
 		cs.ProposalBlockParts = types.NewPartSetFromHeader(proposal.BlockID.PartSetHeader)
 	}
 
-	cs.Logger.Info("received proposal", "proposal", proposal)
```

Comment:
@ancazamfir , before merging this, if you have some spare cycles, could you please double-check your concern about the usage of `GetProposer()` here?

Let me explain why I think it is OK. Look at lines 1887-1891 above. We are using `GetProposer()` to obtain the public key that verifies the proposal's signature. So, if that verification succeeds, the validator returned by `GetProposer()` should be the one that proposed `proposal`. Do you agree?

Reply:
That's a good point, @ancazamfir. I have followed your suggestion, but have moved the log to be hit after the proposer is updated (so we have the proposer corresponding to the round we're entering). I have also added an _Info_ log that will log the proposer in non-zero rounds.
Let me know what you think.